lib/ProductOpener/Tags.pm

```diff
@@ -2541,52 +2543,14 @@ sub canonicalize_taxonomy_tag($$$)
 	$tag =~ s/^ //g;
 	$tag =~ s/ $//g;
 
-	if (($tag =~ /^(\w+:\w\w):(.+)/) and (defined $properties{$tagtype})) {
-		# Test for linked data, ie. wikidata:en:Q1234
-		my $property_key = $1;
-		my $property_value = $2;
-		my $matched_tagid;
-		foreach my $canon_tagid (keys %{$properties{$tagtype}}) {
-			if ((defined $properties{$tagtype}{$canon_tagid}{$property_key}) and ($properties{$tagtype}{$canon_tagid}{$property_key} eq $property_value)) {
-				if (defined $matched_tagid) {
-					# Bail out on multiple matches for a single tag.
-					undef $matched_tagid;
-					last;
-				}
-
-				$matched_tagid = $canon_tagid;
-			}
-		}
-
-		if (defined $matched_tagid) {
-			return $matched_tagid;
-		}
+	my $linked_data_tag = canonicalize_taxonomy_tag_linkeddata($tagtype, $tag);
```

Comment:
In what circumstances would canonicalize_taxonomy_tag() be called with a linked data or weblink identifier?

When they are used in URLs like in http://world.openfoodfacts.org/category/wikidata:en:Q7802 then the canonicalization should already have been done through analyze_request()

canonicalize_taxonomy_tag() is used quite extensively internally, with input values that I think are never wikidata ids, so I would much prefer that we remove any extra subroutine calls from it.

Reply:
I'm not sure about the issue number, but the link detection was built in to `canonicalize_taxonomy_tag` to support users entering Wikidata ids or URLs on the website or the API when adding ie. category tags.

Reply:
I think we should remove that support. I have not seen anyone (or anything) using this. And meanwhile, it adds processing / memory usage for tons of calls to that function, on all taxonomies. E.g. for ingredient parsing, we call canonicalize_taxonomy_tag for each ingredient in the ingredient list. We could keep the support for the URLs (although I'm not really sure who would use it and in what circumstances), but remove for input values to tags.